sycl/source/detail/kernel_compiler/kernel_compiler_sycl.cpp

```diff
@@ -9,59 +9,26 @@
 #include "kernel_compiler_sycl.hpp"
 #include <sycl/exception.hpp> // make_error_code
 
-#if __GNUC__ && __GNUC__ < 8
-
-// std::filesystem is not availalbe for GCC < 8
-// and much of the  cross-platform file handling code depends upon it.
-// Given that this extension is experimental and that the file
-// handling aspects are most likely temporary, it makes sense to
-// simply not support GCC<8.
-
-namespace sycl {
-inline namespace _V1 {
-namespace ext::oneapi::experimental {
-namespace detail {
-
-bool SYCL_Compilation_Available() { return false; }
-
-spirv_vec_t
-SYCL_to_SPIRV(const std::string &SYCLSource, include_pairs_t IncludePairs,
-              const std::vector<std::string> &UserArgs, std::string *LogPtr,
-              const std::vector<std::string> &RegisteredKernelNames) {
-  (void)SYCLSource;
-  (void)IncludePairs;
-  (void)UserArgs;
-  (void)LogPtr;
-  (void)RegisteredKernelNames;
-  throw sycl::exception(sycl::errc::build,
-                        "kernel_compiler does not support GCC<8");
-}
-
-std::string userArgsAsString(const std::vector<std::string> &UserArguments) {
-  return std::accumulate(UserArguments.begin(), UserArguments.end(),
-                         std::string(""),
-                         [](const std::string &A, const std::string &B) {
-                           return A.empty() ? B : A + " " + B;
-                         });
-}
-
-} // namespace detail
-} // namespace ext::oneapi::experimental
-} // namespace _V1
-} // namespace sycl
-
-#else
-
 #include <sycl/detail/os_util.hpp>
 
 #include <ctime>
-#include <filesystem>
 #include <fstream>
 #include <random>
 #include <regex>
 #include <sstream>
 #include <stdio.h>
 
+// For GCC versions less than 8, use experimental/filesystem.
+#if defined(__has_include) && __has_include(<filesystem>)
+#include <filesystem>
+namespace fs = std::filesystem;
+#elif defined(__has_include) && __has_include(<experimental/filesystem>)
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+#else
+#error "kernel_compiler sycl requires C++ filesystem support"
```

Comment:
```suggestion
#error "kernel_compiler requires C++ filesystem support"
```

Reply:
The kernel_compiler requires C++ filesystem support for compiling SYCL C++ code. But it can compile OpenCL C without it.

Reply:
Sorry, I don't understand the meaning of `sycl` in this context. Is this the part of the file name or the name of the standard? 
If this text refers to the name of the standard it must use capital letters.

Where is `kernel_compiler` term defined? Is this some abbreviation of the `sycl_ext_oneapi_kernel_compiler` extension name?